packages/@lwc/engine-core/src/framework/decorators/register.ts

```diff
@@ -74,9 +81,15 @@ function validateObservedField(
 ) {
     if (!isUndefined(descriptor)) {
         const type = getClassDescriptorType(descriptor);
-        assert.fail(
-            `Invalid observed ${fieldName} field. Found a duplicate ${type} with the same name.`
-        );
+        const message = `Invalid observed ${fieldName} field. Found a duplicate ${type} with the same name.`;
+
+        // [W-9927596] Ideally we always throw an error when detecting duplicate observed field.
+        // This branch is only here for backward compatibility reasons.
+        if (type === DescriptorType.Accessor) {
+            logError(message);
```

Comment:
Given this code (notice that it does not have `@api`):

```js
import { LightningElement } from 'lwc';

export default class App extends LightningElement {
    queryFilter = 'some value';

    get queryFilter() {
        return this._queryFilter;
    }

    set queryFilter(value) {
        this._queryFilter = value;
    }
}
```

The old behavior would have thrown. This behavior now will just log an error in the console. I'm good with this change, just pointing a slight difference.

Reply:
I added a [test](https://github.com/salesforce/lwc/pull/2501/files#diff-a2e5eb5d65fd146afd2b3bce8a83c24325f34c80972d530573697cf8b6f5bc78R158-R179) to make sure we don't regress this in future releases.